### PR TITLE
Add info on sync modes to nodes-and-clients docs

### PR DESCRIPTION
**Pull request by taxmeifyoucan**
Elaborating content about synchronization strategies. 

## Description

Sync modes section remade, updated, added more info explaining high level benefits and tradeoffs. 

## Related Issue
Solves #3219

### CODE REVIEW

File: src/content/developers/docs/nodes-and-clients/index.md [hunk 1]
```diff
@@ -155,28 +155,52 @@ Hyperledger Besu is an enterprise-grade Ethereum client for public and permissio
...
-- Fast (Default) – downloads all blocks (including headers, transactions and receipts), verifies all headers, and downloads the state and verifies it against the headers.
-- Light – downloads all block headers, block data, and verifies some randomly.
-- Warp sync – Every 5,000 blocks, nodes will take a consensus-critical snapshot of that block’s state. Any node can fetch these snapshots over the network, enabling a fast sync. [More on Warp](https://openethereum.github.io/Warp-Sync-Snapshot-Format.html)
-- Beam sync – A sync mode that allows you to get going faster. It doesn't require long waits to sync, instead it back-fills data over time. [More on Beam](https://medium.com/@jason.carver/intro-to-beam-sync-a0fd168be14a)
-- Header sync – you can use a trusted checkpoint to start syncing from a more recent header and then leave it up to a background process to fill the gaps eventually
+To follow and verify current data in the network, Ethereum client needs to sync with latest network state. This is done by downloading data from peers, cryptographically verifying their integrity, and building a local blockchain database.
```
Comment (minimalsm) [inline src/content/developers/docs/nodes-and-clients/index.md:160]:
```suggestion
To follow and verify current data in the network, the Ethereum client needs to sync with the latest network state. This is done by downloading data from peers, cryptographically verifying their integrity, and building a local blockchain database. 
```

File: src/content/developers/docs/nodes-and-clients/index.md [hunk 2]
```diff
@@ -155,28 +155,52 @@ Hyperledger Besu is an enterprise-grade Ethereum client for public and permissio
...
-- Beam sync – A sync mode that allows you to get going faster. It doesn't require long waits to sync, instead it back-fills data over time. [More on Beam](https://medium.com/@jason.carver/intro-to-beam-sync-a0fd168be14a)
-- Header sync – you can use a trusted checkpoint to start syncing from a more recent header and then leave it up to a background process to fill the gaps eventually
+To follow and verify current data in the network, Ethereum client needs to sync with latest network state. This is done by downloading data from peers, cryptographically verifying their integrity, and building a local blockchain database. 
 
-You define the type of sync when you get set up, like so:
+Synchronization modes represent different approaches to this process with various trade-offs. Clients also vary in their implementation details of sync algorithms. Always refer to the official documentation of your chosen client for specifics on implementation.
```
Comment (minimalsm) [inline src/content/developers/docs/nodes-and-clients/index.md:162]:
```suggestion
Synchronization modes represent different approaches to this process with various trade-offs. Clients also vary in their implementation of sync algorithms. Always refer to the official documentation of your chosen client for specifics on implementation.
```

File: src/content/developers/docs/nodes-and-clients/index.md [hunk 3]
```diff
@@ -155,28 +155,52 @@ Hyperledger Besu is an enterprise-grade Ethereum client for public and permissio
...
+
+#### Overview of strategies
+
+General overview of synchronization approaches used in mainnet ready clients:
```
Comment (minimalsm) [inline src/content/developers/docs/nodes-and-clients/index.md:166]:
```suggestion
General overview of synchronization approaches used in Mainnet ready clients:
```

File: src/content/developers/docs/nodes-and-clients/index.md [hunk 4]
```diff
@@ -155,28 +155,52 @@ Hyperledger Besu is an enterprise-grade Ethereum client for public and permissio
...
+    - Fastest sync strategy developed by Geth, currently its default
+    - Saves a lot of disk usage and network bandwidth without sacrificing security.
+- Warp sync – Implemented by OpenEthereum. Nodes regularly generate a consensus-critical state snapshot and any peer can fetch these snapshots over the network, enabling a fast sync from this point. [More on Warp](https://openethereum.github.io/Beginner-Introduction#warping---no-warp)
+    - Fastest and default sync mode of OpenEthereum relies on static snapshots served by peers.
+    - Similar strategy as snap sync but without certain security benefits.
+- Beam sync – Implemented by Nethermind and Trinity. Works like fast sync but also downloads the state data (witnesses) for the latest blocks, which allows you to query the chain within the first few minutes from starting. [More on Beam](https://medium.com/@jason.carver/intro-to-beam-sync-a0fd168be14a)
```
Comment (minimalsm) [inline src/content/developers/docs/nodes-and-clients/index.md:183]: I think we should either try to define what a witness is somewhere on this page or remove the reference.

File: src/content/developers/docs/nodes-and-clients/index.md [hunk 5]
```diff
@@ -155,28 +155,52 @@ Hyperledger Besu is an enterprise-grade Ethereum client for public and permissio
...
+    - Syncs state first and enables you to query RPC in a few minutes.
+    - Still in development and not fully relyable, background sync is slowed down and RPC responses might fail.
+
+#### Setup in client
+
+Clients offer rich configuration options to suite your needs. Pick the one which suits you the most based on its level of security, available data, and cost. Apart from the synchronization algorithm, you can also set pruning of different kinds of old data. Pruning enables deleting outdated data, e.g. removing state trie nodes that are unreachable from recent blocks.
```
Comment (minimalsm) [inline src/content/developers/docs/nodes-and-clients/index.md:189]:
```suggestion
Clients offer rich configuration options to suit your needs. Pick the one that suits you best based on the level of security, available data, and cost. Apart from the synchronization algorithm, you can also set pruning of different kinds of old data. Pruning enables deleting outdated data, e.g. removing state trie nodes that are unreachable from recent blocks. 
```

**Review by minimalsm — CHANGES_REQUESTED** [review]
Hey @taxmeifyoucan thanks for the PR on this 💪

Added comments for some details below 👍

**Comment by minimalsm** [comment]
Not sure the best approach but I think the hanging links look a little strange here: 

Screenshot 2021-09-13 at 07 18 34

Maybe we could add it as the last bullet

Screenshot 2021-09-13 at 07 17 41

Or have each sync mode as its own individual sub-heading? i.e:

- Heading
- Description
- Bullet points
- Link

**Comment by minimalsm** [comment]
> Or have each sync mode as its own individual sub-heading? i.e:
> 
> Heading
> Description
> Bullet points
> Link

I've made some changes to implement the above. I think it's a bit cleaner and easier for users to scan.

Screenshot 2021-09-13 at 13 30 25

**Comment by taxmeifyoucan** [comment]
Thank you for proof reading and feedback @minimalsm ! 
I changed the `witnesses` to simple general definition, reader can find more in the attached link anyway. 
It definitely looks better divided into sub-headings like this, perfect!

**Comment by minimalsm** [comment]
Awesome @taxmeifyoucan, thanks for this. Merging now 🙏

**Pull request merged**